ensemble/random_forest.h

```diff
@@ -0,0 +1,61 @@
+#ifndef _MACHINELEARNING_ENSEMBLE_RANDOM_FOREST_H_
+#define _MACHINELEARNING_ENSEMBLE_RANDOM_FOREST_H_
+
+#include "tree/tree.h"
+
+namespace ensemble {
+/**
+ * Base class for Random Forest Classification and Regression algorithms.
+ */
+class RandomForest {
+ public:
+  /**
+   * \brief Standard destructor.
+   */
+  virtual ~RandomForest();
+
+  /**
+   * Compute the out-of-bag score that represents the prediction
+   * for the entire set of samples in the oob.
+   * \return oob score
+   */
+  virtual double OobScore() = 0;
+
+  /**
+   * Compute the out-of-bag score that represents the prediction 
+   * for each sample in the oob.
+   * \param tree_idx Index of the tree
+   * \return oob score
+   */
+  virtual double OobScoreSample(const int tree_idx) = 0;
```

Comment:
Since we are returning the prediction of each sample in the out-of-bog, why the return value here is just a single double? Shouldn't be a vector or something like that, in which contains the index of the document (this is the document in the oob) and its score?